src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/AbstractClassCouplingCheck.java

```diff
@@ -294,7 +304,10 @@ public void checkCoupling() {
          */
         private boolean isSignificant(String candidateClassName) {
             return !excludedClasses.contains(candidateClassName)
-                    && !candidateClassName.startsWith("java.lang.");
+                    && !candidateClassName.startsWith("java.lang.")
+                    && (excludeClassesRegexp == null
+                    || excludeClassesRegexp.isEmpty()
+                    || !candidateClassName.matches(excludeClassesRegexp));
```

Comment:
This will compile regexp each time the execution process enters the method. In order to avoid this you need to compile regexp in setter. A good example is https://github.com/checkstyle/checkstyle/blob/b4a337a599e853cdc44b19a76da28fcfab2956cd/src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyCatchBlockCheck.java#L172.